edge-modules/DirectMethodSender/Program.cs

```diff
@@ -24,21 +26,47 @@ public static async Task<int> MainAsync()
             (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromSeconds(5), Logger);
             DirectMethodSenderBase directMethodClient = null;
             ModuleClient reportClient = null;
+            Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
+            Option<Uri> testReportCoordinatorUrl = Settings.Current.TestResultCoordinatorUrl;
+
             try
             {
+                Guid batchId = Guid.NewGuid();
+                Logger.LogInformation($"Batch Id={batchId}");
+
                 directMethodClient = await CreateClientAsync(Settings.Current.InvocationSource);
 
-                Option<Uri> analyzerUrl = Settings.Current.AnalyzerUrl;
                 reportClient = await ModuleUtil.CreateModuleClientAsync(
                     Settings.Current.TransportType,
                     ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
                     ModuleUtil.DefaultTransientRetryStrategy,
                     Logger);
-                while (!cts.Token.IsCancellationRequested)
+
+                Logger.LogInformation($"Load gen delay start for {Settings.Current.TestStartDelay}.");
+                await Task.Delay(Settings.Current.TestStartDelay);
+
+                DateTime testStartAt = DateTime.UtcNow;
+                while (!cts.Token.IsCancellationRequested && IsTestTimeUp(testStartAt))
                 {
-                    HttpStatusCode result = await directMethodClient.InvokeDirectMethodAsync(cts);
+                    (HttpStatusCode result, long dmCounter) = await directMethodClient.InvokeDirectMethodAsync(cts);
 
                     // TODO: Create an abstract class to handle the reporting client generation
+                    await testReportCoordinatorUrl.ForEachAsync(
+                        async (Uri uri) =>
+                        {
+                            TestResultCoordinatorClient trcClient = new TestResultCoordinatorClient { BaseUrl = uri.AbsoluteUri };
+                            await ModuleUtil.ReportStatus(
+                                    trcClient,
+                                    Logger,
+                                    Settings.Current.ModuleId + ".send",
+                                    ModuleUtil.FormatDirectMethodTestResultValue(
+                                        Settings.Current.TrackingId.Expect(() => new ArgumentException("TrackingId is empty")),
```

Comment:
I think for long haul and e2e test, tracking id is not required.  Why throws argument exception unconditionally if tracking id is missing?